Add docs for shard level stats in node stats (#111082)
elastic/elasticsearch: docs/reference/cluster/nodes-stats.asciidoc

Reviewer: I see that https://www.elastic.co/guide/en/elasticsearch/reference/current/indices-stats.html also can have a shards level that shows these additional statistics. Should we maybe add a small paragraph in that page saying something along the line of `If shards level is requestged, some additional shard-specific statistics are shown, similar to the <<cluster-nodes-stats,nodes stats API>>` ?

Author: That makes sense, it looks like the text describing the level parameter is located [here](https://github.com/elastic/elasticsearch/blob/fd916c221ae51b621dc5eb4d93b8f5a10bbc6d49/docs/reference/rest-api/common-parms.asciidoc?plain=1#L647-L661), so would it be best to add something like that there?
The indices-stats and cluster-nodes-stats are the only doc pages that reference the level parameter it seems

Reviewer: Yes please try that and I'll check the preview (once it's auto-generated) to see if it looks OK.

=======

`shards`::

> (object) Contains statistics aggregated at the `shards` level, per index, as well as shard-specific statistics.

Author: I changed the phrasing just a smidge, hope that is ok

+
.Properties of `shards`
[%collapsible%open]
=======

`<index_name>`::
(array of objects) Contains a list of all shards for the index.
+
.Properties of `<index_name>`
[%collapsible%open]
========

`<shard_id>`::
(object) Contains statistics for the shard.
+
.Properties of `<shard_id>`
[%collapsible%open]
=========

`routing`::
(object) Contains routing information about the shard instance.
+
.Properties of `routing`
[%collapsible%open]
==========

`state`::
(string) State of the shard. Returned values are:
+

> * `INITIALIZING`: The shard is recovering from a peer shard or gateway.

Reviewer: ```suggestion
* `INITIALIZING`: The shard is initializing/recovering.
```

Author: I actually borrowed the phrasing for these stats from [here](https://github.com/elastic/elasticsearch/blob/fd916c221ae51b621dc5eb4d93b8f5a10bbc6d49/docs/reference/cat/shards.asciidoc?plain=1#L72), should I still change it here? (Or change it in both spots?)

Reviewer: I'd prefer what I suggested, or simply `recovering`, just to avoid readers wondering over `gateway`.

* `RELOCATING`: The shard is relocating.
* `STARTED`: The shard has started.
* `UNASSIGNED`: The shard is not assigned to any node.

`primary`::
(Boolean) Whether the shard is a primary shard or not.

`node`::
(string) ID of the node the shard is allocated to.

`relocating_node`::
(string) ID of the node the shard is either relocating to or relocating from, or null if shard is not relocating.

==========

`commit`::
(object) Contains information regarding the last commit point of the shard.
+
.Properties of `commit`
[%collapsible%open]
==========

`id`::
(string) Base64 version of the commit ID.

`generation`::
(integer) Lucene generation of the commit.

`user_data`::
(object) Contains additional technical information about the commit.

`num_docs`::
(integer) The number of docs in the commit.

==========

`seq_no`::

> (object) Contains information about sequence numbers and checkpoints for the shard.

Reviewer: ```suggestion
(object) Contains information about <<optimistic-concurrency-control,sequence numbers>> and checkpoints for the shard.
```

+
.Properties of `seq_no`
[%collapsible%open]
==========

`max_seq_no`::
(integer) The maximum sequence number issued so far.

`local_checkpoint`::
(integer) The current local checkpoint of the shard.

`global_checkpoint`::
(integer) The current global checkpoint of the shard.

==========

`retention_leases`::

> (object) Contains information about history retention leases.

Reviewer: ```suggestion
(object) Contains information about <<index-modules-history-retention,history retention leases>>.
```

+
.Properties of `retention_leases`
[%collapsible%open]
==========

`primary_term`::

> (integer) The current primary term.

Reviewer: ```suggestion
(integer) The primary term of this retention lease collection.
```

`version`::
(integer) The current version of the retention lease collection.

`leases`::
(array of objects) List of current leases for this shard.
+
.Properties of `leases`
[%collapsible%open]
===========

`id`::
(string) The ID of the lease.

`retaining_seq_no`::
(integer) The minimum sequence number to be retained by the lease.

`timestamp`::
(integer) The timestamp of when the lease was created or renewed.
Recorded in milliseconds since the {wikipedia}/Unix_time[Unix Epoch].

`source`::
(string) The source of the lease.

===========
==========

`shard_path`::
(object)
+
.Properties of `shard_path`
[%collapsible%open]
==========

`state_path`::

> (string) The file system path to the shard state.

Reviewer: ```suggestion
(string) The state-path root, without the index name and the shard ID.
```

`data_path`::

> (string) The file system path to the data belonging to the shard.

Reviewer: ```suggestion
(string) The data-path root, without the index name and the shard ID.
```

`is_custom_data_path`::

> (boolean) Whether the data path is a custom set location or not.

Reviewer: ```suggestion
(boolean) Whether the data path is a custom data location and therefore outside of the nodes configured data paths.
```

==========

`search_idle`::

> (boolean) Whether the shard is search idle or not.

Reviewer: ```suggestion
(boolean) Whether the shard is <<dynamic-index-search-idle-after,search idle>> or not.
```
